pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstorecache_test.go

```diff
@@ -0,0 +1,209 @@
+package sigstore
+
+import (
+	"container/list"
+	"reflect"
+	"sync"
+	"testing"
+)
+
+var (
+	selectors1 = Item{
+		Key: "signature1",
+		Value: []SelectorsFromSignatures{
+			{
+
+				Subject:        "spirex1@example.com",
+				Content:        "content1",
+				LogID:          "1111111111111111",
+				IntegratedTime: "1111111111111111",
+			},
+		},
+	}
+
+	selectors2 = Item{
+		Key: "signature2",
+		Value: []SelectorsFromSignatures{
+			{
+
+				Subject:        "spirex2@example.com",
+				Content:        "content2",
+				LogID:          "2222222222222222",
+				IntegratedTime: "2222222222222222",
+			},
+		},
+	}
+
+	selectors3 = Item{
+		Key: "signature3",
+		Value: []SelectorsFromSignatures{
+			{
+				Subject:        "spirex3@example.com",
+				Content:        "content3",
+				LogID:          "3333333333333333",
+				IntegratedTime: "3333333333333333",
+			},
+		},
+	}
+
+	selectors3Updated = Item{
+		Key: "signature3",
+		Value: []SelectorsFromSignatures{
+			{
+				Subject:        "spirex3@example.com",
+				Content:        "content4",
+				LogID:          "4444444444444444",
+				IntegratedTime: "4444444444444444",
+			},
+		},
+	}
+)
+
+func TestNewCache(t *testing.T) {
+	tests := []struct {
```

Comment:
test table driving is an overkill for a single test case, use a single t.Run instead